### PR TITLE
contract tests initial setup

### PR DESCRIPTION
**Pull request by mohitpubnub**
* spike: cucumber-rs integration.
* sample step definition.
* hooks implementation for contract server initialization.

### CODE REVIEW

**Comment by mohitpubnub** [comment]
@Xavrax  Have a look on defining how this workflow will trigger it.
WIP: understanding of managing test suites via cargo.

File: Cargo.toml [hunk 1]
```diff
@@ -1,5 +1,15 @@
 [workspace]
 
+[dev-dependencies]
```
Comment (Xavrax) [inline Cargo.toml:3]: Move it the `crates/contract-tests/Cargo.toml`

File: tests/contract.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
+use cucumber::{ given, when, then, World };
```
Comment (Xavrax) [inline tests/contract.rs:1]: We have contract tests crate - move theses files there. 
`crates/contract-tests/tests/...`

**Review by Xavrax — CHANGES_REQUESTED** [review]

File: tests/contract.rs (hunk 2 above)
Comment (mohitpubnub) [reply tests/contract.rs]: moved

**Review by Xavrax — APPROVED** [review]
if we're fine with the that licenses it LGTM

File: crates/contract-tests/tests/contract.rs [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+                        .find(|&t| t.starts_with("contract="))
+                        .unwrap();
+                    let splitted_values: Vec<&str> = tag.split("=").collect();
+                    if !splitted_values[1].is_empty() {
+                        let script_name = splitted_values[1];
+                        init_server(script_name.to_string()).await;
+                    }
+                }
+            })
+        })
```
Comment (Xavrax) [inline crates/contract-tests/tests/contract.rs:64]:
```suggestion
    PubnubWorld::cucumber()
        .before(|_feature, _rule, scenario, _world| {
            futures::FutureExt::boxed(async move {
                if scenario
                    .tags
                    .iter()
                    .find(|&t| t.starts_with("contract="))
                    .is_some()
                {
                    let tag = _scenario
                        .tags
                        .iter()
                        .find(|&t| t.starts_with("contract="))
                        .unwrap();
                    let splitted_values: Vec<&str> = tag.split("=").collect();
                    if !splitted_values[1].is_empty() {
                        let script_name = splitted_values[1];
                        init_server(script_name.to_string()).await;
                    }
                }
            })
        })
```
Underscore is used for not used variables.

**Pull request merged**